src/sfdxProject.ts

```diff
@@ -248,14 +248,19 @@ export class SfdxProjectJson extends ConfigFile<ConfigFile.Options> {
       return Object.assign({}, packageDir, { name, path, fullPath });
     });
 
-    const defaultDirs = packageDirs.filter((packageDir) => packageDir.default);
-
-    // Don't throw about a missing default path if we are in the global file.
-    // Package directories are not really meant to be set at the global level.
-    if (defaultDirs.length === 0 && !this.isGlobal()) {
-      throw new SfdxError(this.messages.getMessage('MissingDefaultPath'));
-    } else if (defaultDirs.length > 1) {
-      throw new SfdxError(this.messages.getMessage('MultipleDefaultPaths'));
+    // If we only have one package entry, it must be the default even if not explicitly labelled
+    if (packageDirs.length > 1) {
+      const defaultDirs = packageDirs.filter((packageDir) => packageDir.default);
+      // Don't throw about a missing default path if we are in the global file.
+      // Package directories are not really meant to be set at the global level.
+      if (defaultDirs.length === 0 && !this.isGlobal() && packageDirs.length > 1) {
+        throw new SfdxError(this.messages.getMessage('MissingDefaultPath'));
+      } else if (defaultDirs.length > 1) {
+        throw new SfdxError(this.messages.getMessage('MultipleDefaultPaths'));
+      }
+    } else {
+      // add default=true to the package
+      packageDirs[0].default = true;
```

Comment:
Not sure if this is needed - but I don't want to alter consumer's expectations

Reply:
We definitely want to set that if it isn't defined. A couple of things though. 

1. `packageDirs` may be empty here... so you should check for that case. This was handled before by throwing the MissingDefaultPath if the filter came back empty, which would return empty on an empty array.
1. What happens if there is only one `packageDirectories` entry but they set `default: false`?  Do we case? I feel like we should still throw in that case too because they were explicit about saying it wasn't the default. Thoughts?